tsdb/engine/tsm1/engine.go

```diff
@@ -1161,6 +1161,12 @@ func (e *Engine) createVarRefIterator(opt influxql.IteratorOptions, aggregate bo
 				return err
 			}
 
+			if opt.Limit > 0 || opt.Offset > 0 || opt.SLimit > 0 || opt.SOffset > 0 {
```

Comment:
I think there is. We also may want to remove the sort here and instead perform the sort when we're using the tag set to create the iterators inside of the query engine.